### PR TITLE
Implement optimized version of isTextForm

### PR DESCRIPTION
**Pull request by lavjain**
Also, add getObject/setObject to GPDBWritable

### CODE REVIEW

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java [hunk 1]
```diff
@@ -73,6 +71,17 @@
         }
     }
 
+    private static final Map<Integer, Integer> notText = new HashMap<>();
```
Comment (frankgh) [inline server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java]: Another option would be to use a set 
```suggestion
    private static final Set<Integer> notText = new HashSet<>();
```

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java [hunk 2]
```diff
@@ -88,16 +97,18 @@
...
+        return type == null ? UNSUPPORTED_TYPE : type;
     }
 
     public static boolean isArrayType(int OID) {
         DataType type = lookup.get(OID);
         return type != null && type.typeElem != null;
     }
 
+    public static boolean isTextForm(int OID) {
+        return notText.get(OID) == null;
```
Comment (frankgh) [inline server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java]: And check whether the item is in the set
```suggestion
        return !notText.contains(OID);
```

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java [hunk 3]
```diff
@@ -73,6 +71,17 @@
         }
     }
 
+    private static final Map<Integer, Integer> notText = new HashMap<>();
+    static {
+        notText.put(BIGINT.OID, 1);
```
Comment (frankgh) [inline server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java]:
```suggestion
        notText.add(BIGINT.OID);
```

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java [hunk 4]
```diff
@@ -73,6 +71,17 @@
         }
     }
 
+    private static final Map<Integer, Integer> notText = new HashMap<>();
+    static {
```
Comment (frankgh) [inline server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java]: is there any reason we wanted two separate static blocks?

**Comment by lavjain** [comment]
@frankgh Thanks for the review. I have incorporated your suggestions.

File: server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java [hunk 5]
```diff
@@ -59,6 +61,7 @@
     UNSUPPORTED_TYPE(-1);
 
     private static final Map<Integer, DataType> lookup = new HashMap<>();
+    private static final Set<Integer> notText = new HashSet<>();
```
Comment (denalex) [inline server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java]: another alternative is to implement EnumSet which should probably be even better as it uses bitsets for implementation: https://docs.oracle.com/javase/7/docs/api/java/util/EnumSet.html

Comment (frankgh) [reply server/pxf-api/src/main/java/org/greenplum/pxf/api/io/DataType.java]: does EnumSet work with Integers? It seems its geared for enums

**Pull request merged**